### PR TITLE
fix(fast-tests): Count all packages of a module as modified when go.mod changes

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Improve a bit fast unit tests to make sure that we consider all the packages of a module as modified when `go.mod` or `go.sum` changes

When computing impacted packages, when a go.mod file is modified it is better to consider all the packages of a module are modified

### CODE REVIEW

File: tasks/go_test.py [hunk 1]
```diff
@@ -752,6 +754,15 @@ def get_impacted_packages(ctx, build_tags=None):
         if file.endswith(".go") or file.endswith(".mod") or file.endswith(".sum")
     }
 
+    # Modification to go.mod and go.sum should force the tests of the whole module to run
+    for file in files:
+        if file.endswith(".mod") or file.endswith(".sum"):
```
Comment (chouetz) [inline tasks/go_test.py:759]: nit, it's probably not possible to have a `pro.mod` or `oppo.sum` but at least you are sure like this, wdyt?
```suggestion
        if file.endswith("go.mod") or file.endswith("go.sum"):
```

**Review by chouetz — APPROVED** [review]

File: tasks/go_test.py (hunk 1 above)
Comment (amenasria) [reply tasks/go_test.py:759]: Guess we could use `==` at this point

Comment (pgimalac) [reply tasks/go_test.py:759]: I guess `file` is a full path, not just the filename, but I agree checking the file name would be better

Comment (KevinFairise2) [reply tasks/go_test.py:759]: I will apply Nicolas suggestion, assuming no one will ever name its file `coucou.go.mod`

**Pull request merged**